src/olympia/scanners/tasks.py

```diff
@@ -229,18 +229,33 @@ def _run_yara_for_path(scanner_result, path, definition=None):
                     scanner=YARA, is_active=True, definition__isnull=False
                 ).values_list('definition', flat=True)
             )
-
-        rules = yara.compile(source=definition)
+        # Initialize external variables so that compilation works, we'll
+        # override them later when matching.
+        externals = {
```

Comment:
It would be nice if we could export the keys of this dict and add something similar to a `help_text` in the form when creating a Yara rule. I don't know how to do it, though :p

Reply:
Mmm I don't know either but looking at the `models.py` code looks like I should have modified `clean_yara()` as well since it uses `compile()` without referencing the `externals`...

Reply:
oh right..

Reply:
b6414b6a9aed7ebad5f41692436a82678b06c86d fixes that and 8fdfc439e945b8d9e6aa10f88664cac780c4d843 adds a hint (not as a `help_text`, and not DRY, but should be good enough)

Reply:
ah yes, smart. I like it!